README.md

```diff
@@ -1,2 +1,239 @@
-# spectangular
+# Spectangular
+
 Simple Protractor e2e Component Testing for Angular
+
+This library helps with writing e2e tests for Angular applications by supporting the navigation to 
+and interaction with web elements.  The library uses Protactor [Protractor-jasmine](https://angular.github.io/protractor/#/). 
+
+# Protractor version
+
+The current supported Protractor version is 2.1.0.
+ 
+# Demo application
+
+The demo application uses [Angular material](https://material.angularjs.org/latest/#/)
+ 
+[Demo spectangular](https://github.com/mgijsbertihodenpijl/spectangular-demo) 
+
+# Installation
+
+`npm install`
+
+Declare the library as node module into the package.json of your application.
+
+
+# Development (ECMAscript 6)
+
+Because the components are written in ECMAscript 6, the code is transpiled with Babel. During development you can watch 
+and transpile the code automatically into the dist folder.
+ 
+`grunt dev`
+
+# Example 1: login page
+
+This example explains how to write an e2e test for a login page. The login page contains a form with two input fields, one
+for the username and one for the password and a submit button. The page starts on http://localhost:8080/#/inloggen.
+
+There are small differences between ECMAscript 5 and 6 in declaring the spectangular library 
+
+## ECMAscript 6 
+<pre><code class="language-javascript" >
+    import Spectangular from '../node_modules/spectangular/dist/spectangular.js';
+    import * as SpectangularMdLibrary from '../node_modules/spectangular/dist/libraries/md/md.js';
+</code></pre>
+
+## ECMAscript 5 
+<pre><code class="language-javascript" >
+    var Spectangular = require('../node_modules/spectangular/dist/spectangular.js'');
+    var SpectangularMdLibrary = require('../node_modules/spectangular/dist/libraries/md/md.js');
+</code></pre>
+
+
+The example (ECMA6script) :  
+
+<pre><code class="language-javascript" >
+    import Spectangular from '../node_modules/spectangular/dist/spectangular.js';
+    import * as SpectangularMdLibrary from '../node_modules/spectangular/dist/libraries/md/md.js';
+
+    Spectangular.baseUrl = 'http://localhost:8080';
+    Spectangular.library = SpectangularMdLibrary
+
+    //loads the startpage        
+    Spectangular.loadPage('/#/start/', 'input#username');
+           
+    //selectors and data for the form       
+    var buttonSelector = 'material-button.material-success';    
+    var formData = [
+    {"model": "user.username", "value": "admin", "type": "textfield"},
+    {"model": "user.password", "value": "welcome", "type": "textfield"}
+    ]
+    
+    //execution
+   Spectangular.form({
+    selector: '[name=\"userForm\"]',
+    data: formData
+    })
+   
+    Spectangular.button({css: buttonSelector}).click();
```

Comment:
Ja maar ik maak me niet zo druk om de performance van een e2e test. Helemaal niet aangezien een css selector en array filter tegenwoordig zo performant is.

On 21 Jul 2015, at 11:19, Martijn Gijsberti Hodenpijl <notifications@github.com> wrote:

In README.md <https://github.com/schubergphilis/spectangular/pull/1#discussion_r35083927>:

> +    Spectangular.loadPage('/#/start/', 'input#username');
> +           
> +    //selectors and data for the form       
> +    var buttonSelector = 'material-button.material-success';    
> +    var formData = [
> +    {"model": "user.username", "value": "admin", "type": "textfield"},
> +    {"model": "user.password", "value": "welcome", "type": "textfield"}
> +    ]
> +    
> +    //execution
> +   Spectangular.form({
> +    selector: '[name=\"userForm\"]',
> +    data: formData
> +    })
> +   
> +    Spectangular.button({css: buttonSelector}).click();
Overigens is met text wel trager dan een specifieke selector. By text haalt alle buttons op van een page, en doet dan het filter.

—
Reply to this email directly or view it on GitHub <https://github.com/schubergphilis/spectangular/pull/1/files#r35083927>.